### DISCUSSION TITLE
Use on a specific page of web application?

### DISCUSSION THREAD

**Discussion by ShrineFox** [Hawxy/Discord.Addons.Hosting#21] [Q&A] [answered]
Hi, first of all this is brilliant and thank you for making it. 
I'm new to MVC web applications, got the bot running but I'm curious if it's possible to only launch the bot from a specific url so I can continue serving cshtml from other pages. As it is, any page never finishes loading while the bot is running. I'd be fine with that happening on a single dedicated page, even better if I can still serve content while it runs, but I understand if this just wasn't meant to be used that way.

**Comment by Hawxy** [comment]
Could you provide an explanation of what the architecture is here? The bot runs in the background and doesn't involve itself with the web pipeline at all. This kinda sounds like you're using a host that starts and stops the host for the lifetime of the web request.

**Comment by ShrineFox** [comment, marked as answer]
Ahh okay, I was hoping to hear that should be the case normally. Haven't tested on live servers yet, right now this through Visual Studio's IIS on Windows 10. Unless that has to do with it I think I might've configured something wrong then, but not sure what.

~~I have made a [sample project](https://github.com/ShrineFox/AspDotNetCore5.0Razor/commit/6f33e69cc3dd6ea869ee1a4eb4923ae953e8c613) to illustrate what's going on.
First I created a new ASP.NET Core 5.0 project, this time chose Razor Pages instead of MVC (same outcome).
Then I overwrote the default project files with [SampleBotSimple](https://github.com/Hawxy/Discord.Addons.Hosting/tree/master/Samples/SampleBotSimple) from your repository, changing the token and prefix in appsettings.json.
I have it set up so you can test it yourself if you'd like, in Release mode it uses the original HostBuilder which finishes loading a page without any of the bot stuff. If you switch to Debug, the bot comes online and responds to commands, but nothing finishes loading in the browser.~~

I appreciate the quick response, thanks for your time 

EDIT: Wait, I think maybe it's because I missed the background worker step lol, I'll update this comment and close it if that's the case. Testing right now

**Reply by Hawxy** [reply]
Sample is still private :)

**Reply by ShrineFox** [reply]
Whoops, made it public now but talking through it helped me figure out the issue. I misunderstood the directions before cuz I didn't notice creating a background worker makes a whole separate project. After replacing the proper files and adding the token to client secret, it seems to work independently now. 

Thanks for baring with me here, will be in touch if I have any other (real) issues, but I think this will serve me well at this point. Kudos!

**Reply by Hawxy** [reply]
No problem. Looking at the sample it's pretty clear what happened. If you're integrating into ASP.NET Core all you need to do is append the discord-specific extensions (`ConfigureDiscordHost`/`UseCommandService`) to the HostBuilder, not replace the `.ConfigureWebHostDefaults` call. 

ie:

```cs
  Host.CreateDefaultBuilder(args)
                .ConfigureDiscordHost((context, config) =>
                {
                    config.SocketConfig = new DiscordSocketConfig
                    {
                        LogLevel = LogSeverity.Verbose,
                        AlwaysDownloadUsers = true,
                        MessageCacheSize = 200
                    };
                    config.Token = context.Configuration["Token"];
                })
                //Omit this if you don't use the command service
                .UseCommandService((context, config) =>
                {
                    config.DefaultRunMode = RunMode.Async;
                    config.CaseSensitiveCommands = false;
                }) 
                .ConfigureWebHostDefaults(webBuilder =>
                {
                    webBuilder.UseStartup<Startup>();
                });
```
The worker service is the default as the majority of people don't need a web server to go with the bot.

In the next release I'll add a ASP.NET Core example so others don't run into this.
